README.md

```diff
@@ -27,11 +31,70 @@ module Api
 end
 ```
 
-# RSpec configuration
+## Basic usage
+
+### Permitted parameter handling
+Use [jsonapi_parameters](https://github.com/visualitypl/jsonapi_parameters) to transform incoming JSON:API compliant data into common Rails parameters
+```ruby
+def permitted_params
+  params.from_jsonapi
+        .require(:user)
+        .permit(:first_name, :last_name)
+end
+```
+
+### Responding
+Use `respond_with` to initiate transformation(serialization) of domain objects into HTTP response.
+```ruby
+def show
+  respond_with User.find(params[:id])
+end
+```
 
-spec/support/infinum_json_api_setup.rb
+`respond_with` is well integrated with `ActiveRecord::Model` interface. Given a compliant object, the method will correctly set a response status and handle object(or error) serialization based on the presence of `.errors`. For a successful domain operation, HTTP status will be 200 OK(or 204 in case of `create` controller action). Unsuccessful operations will have HTTP status 422 Unprocessable Entity with errors structured according to [JSON:API specification](https://jsonapi.org/format/#error-objects).
+```ruby
+def create
+  respond_with User.create(permitted_params)
+end
+```
 
+`respond_with` also detects usage from a `destroy` controller action and responds with HTTP status 200 OK and an empty body.
+```ruby
+def destroy
+  respond_with User.destroy(params[:id])
+end
 ```
+
+## Internals
+This section explains the under-the-hood behavior of the library.
+
+### Content negotiation
+`InfinumJsonApiSetup::JsonApi::ContentNegotiation` module is designed to integrate [server responsibilities](https://jsonapi.org/format/#content-negotiation-servers) of content negotiation protocol described by the JSON:API specification.
+
+### Error handling
+`InfinumJsonApiSetup::JsonApi::ErrorHandling` module is designed to catch and handle common exceptions that might bubble up when processing a request.
+
+| Exception                                      | HTTP status | Bugsnag notification |
+| ---                                            | :---:       | :---:                |
+| `ActionController::ParameterMissing`           | 400         |                      |
+| `ActionDispatch::Http::Parameters::ParseError` | 400         |                      |
+| `Jure::UnpermittedSortParameters`              | 400         | :white_check_mark:   |
```

Comment:
Jure? 😂

Reply:
Yep, his deep nesting negatively impacts my table layout, so I decided to plant an easter egg. Kind of like it tbh...